### ISSUE TITLE
The top 12 fragments limitation

### ISSUE THREAD

**Issue by Chen-micslab** [vdemichev/DiaNN#802]
Hello Vadim,

The output library of DIA-NN only contains a maximum of 12 fragments percursor. Does it mean that even if a precursor in input library contains more than 12 fragments, DIA-NN can only use top12 of them? Can I modify this parameter? Change it to a number exceeding 12?

Best regards

Chen

**Comment by michaelsteidel86** [comment]
Have tested this command described in the docs?

--max-fr 
(„specifies the maximum number of fragments per precursors in the spectral library being saved“)

Best
Michael

**Comment by vdemichev** [comment]
Hi Chen,

The predicted library by default is indeed just 12 fragments.
All fragments are used from the input library, it's just when using a predicted lib it makes very little benefit to use fragments beyond top 12.

Best,
Vadim

**Comment by Chen-micslab** [comment]
Does this mean that my input library contains 20 fragments of a precursor, all the fragments  will be used for scoring? Then can I display the matching of all 20 fragments in the reports file by modifying the -- max-fr parameter to 20?

**Comment by vdemichev** [comment]
Yes, all fragments are used for scoring. The default for --max-fr is infinity, i.e. all fragments are used, --max-fr 20 will just restrict this number to <= 20. In general, fragments beyond top 12 contribute very little to scoring.

**Comment by Chen-micslab** [comment]
When I set the --max-fr to 20, the 'Fragment.Quant.Raw' in the main report is still only containing 12 fragments, what is the reason?

**Comment by vdemichev** [comment]
This setting does not affect number of fragments used for quantitation.

**Comment by Chen-micslab** [comment]
But I want to see the top 20 fragments in 'Fragment.Quant.Raw', how should I set the parameters?

**Comment by Chen-micslab** [comment]
Could you please guide me how to change the number of fragments in  'Fragment.Quant.Raw'? Thank you very much!

**Comment by vdemichev** [comment]
It's not possible to increase it, this is hard-coded in DIA-NN code.

**Comment by Chen-micslab** [comment]
Thank you very much for response. I have another question. When I choose the 'generate spectral library', each peptide in the generated library has the same fragment ion intensity as the input spectral library. Could DIA-NN generate a library based on the intensity of fragments in  DIA data?

**Comment by cctsou** [comment]
Hi @vdemichev , I have a related question about the number of fragments, what determines how many fragments are used for quant? If I draw a histogram of fragment numbers in the report.tsv there is a spike at 6 and 12, I can understand 12 because the library was capped at 12 fragments. What about the spike at 6, is it because some peptides use top-6 fragments some don't?

![image](https://github.com/vdemichev/DiaNN/assets/3207894/4e68a3cf-760a-4779-9a17-533c67b08d06)
